homeassistant/components/versasense/sensor.py

```diff
@@ -0,0 +1,93 @@
+"""Support for VersaSense sensor peripheral."""
+import logging
+
+from homeassistant.helpers.entity import Entity
+
+from . import DOMAIN
+from .const import (
+    KEY_IDENTIFIER,
+    KEY_PARENT_NAME,
+    KEY_PARENT_MAC,
+    KEY_UNIT,
+    KEY_MEASUREMENT,
+    KEY_CONSUMER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the sensor platform."""
+    consumer = hass.data[DOMAIN][KEY_CONSUMER]
```

Comment:
Please add a guard clause here that checks if discovery_info is None and return if so.

This will guard against the platform getting set up via a platform config section.